modules/gapi/src/backends/cpu/gcpuvideo.cpp

```diff
@@ -107,13 +107,59 @@ GAPI_OCV_KERNEL_ST(GCPUBackgroundSubtractor,
     }
 };
 
+GAPI_OCV_KERNEL_ST(GCPUKalmanFilter, cv::gapi::video::GKalmanFilter, cv::KalmanFilter)
+{
+    static void setup(const cv::GMatDesc&, const cv::GOpaqueDesc&,
+                      const cv::GMatDesc&, const cv::gapi::video::KalmanParams& kfParams,
+                      std::shared_ptr<cv::KalmanFilter> &state, const cv::GCompileArgs&)
+    {
+        GAPI_Assert(kfParams.type == CV_32F || kfParams.type == CV_64F);
+        state = std::make_shared<cv::KalmanFilter>(kfParams.dpDim, kfParams.mpDim, kfParams.ctrlDim, kfParams.type);
+
+        // initial state
+        state->statePre = kfParams.statePre;
+        state->errorCovPre = kfParams.errorCovPre;
+
+        // dynamic system initialization
+        state->controlMatrix = kfParams.controlMatrix;
+        state->measurementMatrix = kfParams.measurementMatrix;
+
+        if (cv::norm(kfParams.transitionMatrix, cv::NORM_INF) != 0)
+            state->transitionMatrix = kfParams.transitionMatrix;
+
+        if (cv::norm(kfParams.processNoiseCov, cv::NORM_INF) != 0)
+            state->processNoiseCov = kfParams.processNoiseCov;
+
+        if (cv::norm(kfParams.measurementNoiseCov, cv::NORM_INF) != 0)
+            state->measurementNoiseCov = kfParams.measurementNoiseCov;
+    }
+
+    static void run(const cv::Mat& measurements, bool haveMeasurement,
+                    const cv::Mat& control, const cv::gapi::video::KalmanParams&,
+                    cv::Mat &out, cv::KalmanFilter& state)
+    {
+        cv::Mat pre;
+
+        if (!control.empty() && cv::norm(control, cv::NORM_INF) != 0)
+            pre = state.predict(control);
+        else
+            pre = state.predict();
+
+        if (haveMeasurement && !measurements.empty())
```

Comment:
Can measurements be passed to our kernel empty? I mean should we check this?

Reply:
Removed.